porogo/api/models.py

```diff
@@ -1,3 +1,122 @@
+'''Define modules used by the backend'''
 from django.db import models
 
-# Create your models here.
+class URL(models.Model):
+    """
+    Stores URLs for many to many relationships
+    
+    Attributes:
+        url (string): URL to store
+    """
+    url = models.URLField(max_length=200)
+
+def get_dest_path(_, filename):
+    """
+    Get path for a given file based on its filename
+    """
+    filename = filename.replace(" ", "")
+    index_of_dash = filename.find('-')
+
+    if index_of_dash == -1:
+        return f'media/other/{filename}' 
+    return f'media/{filename[None:index_of_dash]}/{filename}'
+
+class Image(models.Model):
+    """
+    Stores URLs for many to many relationships
+    
+    Attributes:
+        url (string): URL to store
+    """
+    img = models.FileField(upload_to=get_dest_path)
+    desc = models.CharField(max_length=32)
+
+class Language(models.Model):
+    """
+    Stores Languages for many to many relationships
+    
+    Attributes:
+        name (string): language name to store
+    """
+    name = models.CharField(max_length=16, unique=True)
+
+class Developer(models.Model):
+    """
+    Stores Developers on the site
+
+    Attributes:
+        first_name (str): First name of dev
+        last_name (str): Last name of dev
+        email (str): Email of dev
+        short_description (str): Description of the dev
+        description_experience (str): Description of the dev's experience
+        description_personal (str): Description of the dev's personal interests
+        github_name (str): Github username of dev
+        linkedin_url (str): Linkedin url for the dev
+        other_url (str): Url to another site
+        other_url_type (str): Type of the other url
+        picture (File): Profile picture of the dev
+        featured_project (Project): Project that the user wants to display
+    """
+    first_name = models.CharField(max_length=20)
+    last_name = models.CharField(max_length=20)
+    email = models.EmailField()
+    short_description = models.TextField(max_length=512)
+    description_experience = models.TextField(max_length=1024)
+    description_personal = models.TextField(max_length=1024)
+    github_name = models.CharField(max_length=20, unique=True)
+    linkedin_url = models.URLField(max_length=200, null=True)
+    other_url = models.URLField(max_length=200, null=True)
+    other_url_type = models.CharField(max_length=20, null=True)
+    picture = models.ForeignKey(Image, on_delete=models.CASCADE)
```

Comment:
Wouldn't there only be one picture of the dev? Unless we make some sort of carrousel.

Reply:
Since it's marked as ForeignKey there will only be one. We could've used a FileField directly but I added a description to put in the alt text when you get the image so we should use the Image table.

Reply:
Alright that makes sense